### PR TITLE
SDIT-548 Add services to support upcoming ReleasePrisonerService

### PR DESCRIPTION
**Pull request by mikehalmamoj**

### CODE REVIEW

File: src/main/java/uk/gov/justice/hmpps/prison/service/enteringandleaving/CaseNoteMovementService.kt [hunk 1]
```diff
@@ -25,16 +26,34 @@ class CaseNoteMovementService(
...
+
+  fun createReleaseNote(booking: OffenderBooking, movement: ExternalMovement) {
+    createMovementCaseNote(
+      booking = booking,
+      typeCode = "PRISON",
+      subTypeCode = "RELEASE",
+      note = "Released from ${movement.fromAgency.description} for reason: ${movement.movementReason.description}.",
+      movementTime = movement.movementTime,
+    )
+  }
```
Comment (mikehalmamoj) [inline src/main/java/uk/gov/justice/hmpps/prison/service/enteringandleaving/CaseNoteMovementService.kt:46]: First new service

File: src/main/java/uk/gov/justice/hmpps/prison/repository/jpa/repository/MovementTypeAndReasonRepository.java [hunk 2]
```diff
@@ -5,6 +5,6 @@
 
 import java.util.List;
 
-public interface MovementTypeAndReasonRespository extends CrudRepository<MovementTypeAndReason, MovementTypeAndReason.Pk> {
+public interface MovementTypeAndReasonRepository extends CrudRepository<MovementTypeAndReason, MovementTypeAndReason.Pk> {
```
Comment (mikehalmamoj) [inline src/main/java/uk/gov/justice/hmpps/prison/repository/jpa/repository/MovementTypeAndReasonRepository.java:8]: Fix old typo

File: src/main/java/uk/gov/justice/hmpps/prison/service/enteringandleaving/ExternalMovementService.kt [hunk 3]
```diff
@@ -187,7 +213,7 @@ class ExternalMovementService(
       ?.let { success(it) }
       ?: failure(EntityNotFoundException.withMessage("No movement reason $movementReasonCode found"))
 
-  fun getReceiveDateTime(movementTime: LocalDateTime?, booking: OffenderBooking?): Result<LocalDateTime> {
+  fun getMovementDateTime(movementTime: LocalDateTime?, booking: OffenderBooking?): Result<LocalDateTime> {
```
Comment (mikehalmamoj) [inline src/main/java/uk/gov/justice/hmpps/prison/service/enteringandleaving/ExternalMovementService.kt:216]: Renamed as it's more generic than advertised

File: src/main/java/uk/gov/justice/hmpps/prison/service/enteringandleaving/ExternalMovementService.kt [hunk 4]
```diff
@@ -28,13 +28,13 @@ class ExternalMovementService(
   private val movementTypeRepository: ReferenceCodeRepository<MovementType>,
   private val entityManager: EntityManager,
 ) {
-  fun updateMovementsForTransfer(
+  fun updateMovementsForTransferIn(
```
Comment (mikehalmamoj) [inline src/main/java/uk/gov/justice/hmpps/prison/service/enteringandleaving/ExternalMovementService.kt:31]: Renamed to be more precise

**Review by petergphillips — APPROVED** [review]
LGTM

**Pull request merged**